### PR TITLE
Show outline per tab in FB type editor

### PR DESCRIPTION
**Pull request by mx990**
Show an outline per tab in the FB type editor. This is necessary to show a specific outline for the ST editor and also simplifies outlines for CFB and Subapp editors. It uses an outline page that itself switches between different pages for each tab like the *Outline* view itself, while also handling selections and menu/toolbar contributions.

### CODE REVIEW

File: plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageEditorContentOutlinePage.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	public void setFocus() {
+		if (book != null && !book.isDisposed()) {
+			book.setFocus();
+		}
+		if (currentPage != null) {
+			currentPage.page().setFocus();
+		}
+	}
+
+	public <T> T getAdapter(final Class<T> key) {
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageEditorContentOutlinePage.java:227]: this method is not used. shell the outlinepage implement adaptable?

File: plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageSite.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+		}
+		PartSite.registerContextMenu(menuID, menuMgr, selProvider, false, partSite.getPart(), e4Context, menuExtenders);
+	}
+
+	@Override
+	public void setSelectionProvider(final ISelectionProvider provider) {
+		selectionProvider = provider;
+	}
+
+	/* Package */IEclipseContext getSiteContext() {
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageSite.java]: what is the reason for this comment?

File: plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageEditorContentOutlinePage.java (hunk 1 above)
Comment (mx990) [reply plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageEditorContentOutlinePage.java:227]: Ah, yes, thanks for noticing!

File: plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageSite.java (hunk 2 above)
Comment (mx990) [reply plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageSite.java]: That was a leftover from the original `PageSite`, removed now.

**Review by azoitl — APPROVED** [review]

**Pull request merged**